builtins/amp-ad.js

```diff
@@ -284,6 +285,28 @@ export function installAd(win) {
       return loadPromise(this.iframe_);
     }
 
+    /** @override  */
+    unlayoutCallback() {
+      if (this.iframe_) {
+        removeElement(this.iframe_);
+        if (this.placeholder_) {
+          this.togglePlaceholder(true);
+        }
+        if (this.fallback_) {
+          this.toggleFallback(false);
+        }
+
+        this.iframe_ = null;
+        this.intersectionObserver_ = null;
+      }
+      return true;
+    }
+
+    /** @override  */
+    unlayoutOnPause() {
+      return true;
```

Comment:
Cool. This sounds very good then. But to revert quickly back: what are we getting for unlayout on PAUSE? Why not wait for INACTIVE? Not saying we should or shouldn't - just want to get the complete picture.